instrumentation/elasticsearch/src/main/scala/net/cakesolutions/testkit/logging/elasticsearch/ElasticSearchLogSource.scala

```diff
@@ -2,28 +2,69 @@
 
 package net.cakesolutions.testkit.logging.elasticsearch
 
-import scala.concurrent.Promise
+import java.time.{Instant, ZonedDateTime, ZoneOffset}
 
+import scala.concurrent.Future
+import scala.collection.JavaConverters._
+import com.amazonaws.services.logs.AWSLogsAsyncClientBuilder
+import com.amazonaws.services.logs.model.GetLogEventsRequest
+import com.typesafe.scalalogging.Logger
 import io.circe.Json
-import monix.execution.Scheduler
-import monix.reactive.observers.Subscriber
-
+import io.circe.parser.parse
+import monix.execution.{Cancelable, Scheduler}
+import monix.reactive.{Observable, OverflowStrategy}
+import net.cakesolutions.testkit.config.Configuration.Logging
 import net.cakesolutions.testkit.logging.{LogEvent, LoggingSource}
 
-// $COVERAGE-OFF$ disabled until this is implemented
+import scala.util.control.NonFatal
+
+class ElasticSearchLogSource(id: String, logStreamName: String)
```

Comment:
Here, maybe `id` and `logStreamName` should be treated as the same thing?

I'd also suggest that we might want to pass in a query parameter here as well? I'm thinking it might be a good idea to filter out noise prior to use parsing log lines.